lwcomponent/catalog.go

```diff
@@ -279,63 +278,69 @@ func NewCatalog(
 			}
 		}
 
-		api := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		apiInfo := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		cdkComponents[c.Name] = NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), apiInfo, nil)
+	}
 
-		host, found := localComponents[c.Name]
-		if found {
-			delete(localComponents, c.Name)
-		}
+	components, err := LoadComponents(cdkComponents)
+	if err != nil {
+		return nil, err
+	}
 
-		component := NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), api, host)
+	return &Catalog{client, components, stageConstructor}, nil
+}
 
-		cdkComponents[c.Name] = component
+func NewCachedCatalog(
+	client *api.Client,
+	stageConstructor StageConstructor,
+	cachedComponentsApiInfo map[string]ApiInfo,
+) (*Catalog, error) {
+	if stageConstructor == nil {
+		return nil, errors.New("nil Catalog StageConstructor")
 	}
 
-	for _, localHost := range localComponents {
-		if localHost.Development() {
-			devInfo, err := NewDevInfo(localHost.Dir())
-			if err != nil {
-				return nil, err
-			}
+	cachedComponents := make(map[string]CDKComponent, len(cachedComponentsApiInfo))
 
-			cdkComponents[localHost.Name()] = NewCDKComponent(
-				localHost.Name(),
-				devInfo.Desc,
-				devInfo.ComponentType,
-				nil,
-				localHost)
-		} else {
-			// @jon-stewart: persisted API info
-			cdkComponents[localHost.Name()] = NewCDKComponent(localHost.Name(), "", BinaryType, nil, localHost)
-		}
+	for _, a := range cachedComponentsApiInfo {
+		a := a
```

Comment:
uhmm what are we trying to do here? this assignation looks confusing and dangerous, definitely some unit tests are required to avoid us (reviewers) to test many permutations here.

Reply:
Updated `cachedComponentsApiInfo` to be a map of pointers instead of copies.

Had `a := a` to declare a local copy to make `&a` not point to the value of last item. LOL